src/repositoryModals.jsx

```diff
@@ -0,0 +1,451 @@
+    refreshRemotes: PropTypes.func.isRequired,
+};
+
+export const EditRepositoryModal = ({ remote, availableRemotes }) => {
+    const Dialogs = useDialogs();
+    const [addAnotherKey, setAddAnotherKey] = useState(false);
+    const [key, setKey] = useState('');
+    const [error, setError] = useState('');
+    const [isSelectRemoteExpanded, setSelectRemoteExpanded] = useState(false);
+    const [selectedRemote, setSelectedRemote] = useState(remote);
+    const [newURL, setNewURL] = useState('');
+    const [isTrusted, setIsTrusted] = useState(null);
+
+    useEffect(() => {
+        remotes.loadRemoteSettings(selectedRemote)
+                .then(remoteSettings => {
+                    setNewURL(remoteSettings.url);
+                    setIsTrusted(remoteSettings['gpg-verify'] === "true");
+                });
+    }, [selectedRemote]);
+
+    if (!newURL)
+        return;
+
+    const onUpdate = () => {
+        const promises = [];
+        if (key)
+            promises.push(remotes.importGPGKey(selectedRemote, key));
+
+        const options = {
+            url: newURL,
+            "gpg-verify": isTrusted,
+        };
+
+        promises.push(remotes.updateRemoteSettings(selectedRemote, options));
+
+        Promise.all(promises).then(() => Dialogs.close(), ex => setError(ex.message));
+    };
+
+    const actions = [
+        <Button key="edit-repo" variant="primary" onClick={onUpdate}>
+            {_("Save")}
+        </Button>,
+        <Button key="cancel" variant="link" onClick={Dialogs.close}>
+            {_("Cancel")}
+        </Button>,
+    ];
+
+    return (
+        <Modal title={_("Edit repository")}
+            position="top"
+            variant="small"
+            id="edit-repository-modal"
+            isOpen
+            onClose={Dialogs.close}
+            actions={actions}
+        >
+            <Form isHorizontal>
+                {error && <Alert variant="danger" isInline title={error} />}
+                <FormGroup label={_("Repository")}
+                    id="select-repository"
+                    fieldId="select-repository"
+                >
+                    <Select isOpen={isSelectRemoteExpanded}
+                        selections={selectedRemote}
+                        onToggle={(_, expanded) => setSelectRemoteExpanded(expanded) }
+                        onSelect={(_, remote) => { setSelectRemoteExpanded(false); setSelectedRemote(remote) }}
+                        menuAppendTo="parent"
+                    >
+                        {availableRemotes.map(remote => <SelectOption id={remote} value={remote} key={remote}> {remote} </SelectOption>)}
+                    </Select>
+                </FormGroup>
+                <FormGroup label={_("URL")}
+                    fieldId="edit-remote-url"
+                >
+                    <TextInput id="edit-remote-url"
+                        value={newURL}
+                        onChange={(_ev, url) => setNewURL(url)}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-403-20231128-135012-30a3544e-fedora-coreos-devel//Coverage/lcov/github-pr.diff.gcov.html)